src/System.Private.CoreLib/shared/System/DefaultBinder.cs

```diff
@@ -1204,6 +1204,63 @@ private static bool CreateParamOrder(int[] paramOrder, ParameterInfo[] pars, str
             return true;
         }
 
+        // CanChangePrimitive
+        // This will determine if the source can be converted to the target type
+        private static bool CanChangePrimitive(Type source, Type target)
+        {
+            Primitives widerCodes = s_primitiveConversions[(int)(Type.GetTypeCode(source))];
+            Primitives targetCode = (Primitives)(1 << (int)(Type.GetTypeCode(target)));
+
+            return (widerCodes & targetCode) != 0;
+        }
+
+        private static bool CanChangePrimitiveObjectToType(object source, Type type)
+        {
+            return source == null || CanChangePrimitive(source.GetType(), type);
```

Comment:
`source == null` returns true because native impl used to do it https://github.com/dotnet/coreclr/blob/d3e39bc2f81e3dbf9e4b96347f62b49d8700336c/src/vm/reflectioninvocation.cpp#L2913-L2914

Reply:
None of the callers passes in null here. This null check can be deleted, I think.

Reply:
Removed. Also removed the whole method too (replaced with explicit `CanChangePrimitive`).